crates/sui-framework/packages/sui-framework/sources/coin.move

```diff
@@ -272,9 +283,44 @@ public fun create_regulated_currency_v2<T: drop>(
         coin_metadata_object: object::id(&metadata),
         deny_cap_object: object::id(&deny_cap),
     });
+    if (ctx.epoch() >= REGULATED_MARKER_START_EPOCH) {
```

Comment:
Why do we need to do epoch checks on creation / migration?

Reply:
It's a fair question--I guess there's no harm to adding the marker right away. I will do that instead.

Reply:
Reworked a lot of stuff based on this--now using an enum that gets added to every metadata (not jus the regulated one), and no more epoch checks. Thanks!